### PR TITLE
Fix/story export serialization

### LINKED ISSUE #610: Optimze various `Story.to_dict` methods
Currently, in the `Story` class, there are various `to_dict()` methods.

https://github.com/taranis-ai/taranis-ai/blob/1b0e51dfa13d2cc93163c5d0c0e74e15739eaf5b/src/core/core/model/story.py#L1116-L1137

From analysis with @b3n4kh we think:
- Line 1124 and 1125 are obsolete and should just be removed.
-  Line 1132: `to_worker_dict` uses news_item.to_dict() which would be the difference to `news_items.to_detail_dict` in `to_dict` and `to_detail_dict`.
- Line 1133-1135 need to be inspected if they are needed and could be removed, and essentially they could just be replaced by using the same format as in detail_dict.

slightly related to #609

---

### Maintenance update (2026-07-01)

- Acceptance: identify the current `Story.to_dict` variants, remove duplicate work where measurable, and preserve response shapes used by core/frontend callers.
- Validation: focused core tests for affected serializers plus a small before/after timing note if performance is the reason for the change.

### PR DESCRIPTION
**Pull request by b3n4kh**
Fixes: #610

This pull request introduces a comprehensive overhaul and enhancement of the story export functionality for both administrators and Assess users. The changes add a new API for exporting selected stories with access control, improve export metadata, optimize performance, update documentation, and expand test coverage for edge cases and validation.

**Key changes include:**

### Story Export Functionality

- **New API endpoint for exporting selected stories:** Introduces a `/api/assess/stories/export` endpoint that allows users with `ASSESS_ACCESS` to export a selection of stories as an import-compatible JSON download. The export enforces read access and TLP checks for all stories and their linked news items, returning a 404 if any selection is inaccessible. [[1]](diffhunk://#diff-335ec59704a24fd32453f4cc990498d4fa2dffff6ed61522beacb90d2d6bd407R252-R261) [[2]](diffhunk://#diff-263f8f989b444b99125de1442d149b41155dc4c78070f696c5b30a25e99d60dfL154-R194) [[3]](diffhunk://#diff-365143c9eaa6771777c296b844a7d6fcba740f8914dd4c0a43b6ee11b9ab9daaR6138-R6169)
- **Admin story export enhancements:** The admin "Export Stories" endpoint now supports date filtering with inclusive UTC bounds, explicit metadata toggling, and improved error handling for invalid, future, or reversed date ranges. Exported data includes story attributes and improved date normalization. [[1]](diffhunk://#diff-0c383d209f5b29a35710ee458bc86a4f95dca65def16b6cc97c2c1149efc3570L25-R26) [[2]](diffhunk://#diff-365143c9eaa6771777c296b844a7d6fcba740f8914dd4c0a43b6ee11b9ab9daaL8689-R8743)

### Data Model and Serialization

- **Dedicated export serializer:** Adds `Story.to_export_dict()`, which includes story attributes but omits UI-only and detail fields, ensuring efficient and consistent export payloads. News item arrays reflect the current local display order. [[1]](diffhunk://#diff-8da95a12d8f534dcd0c1de4c2bef2ab009f64cad53b62b1dcaabda85717cff08L1494-R1504) [[2]](diffhunk://#diff-263f8f989b444b99125de1442d149b41155dc4c78070f696c5b30a25e99d60dfL115-R118)
- **Optimized relationship loading:** Batched loading of related story/news item attributes, tags, and sources reduces SQL queries and improves export performance.

### Documentation

- **Extensive documentation updates:** New and updated markdown files (`story-export.md`, `story-exports.md`) detail export formats, access checks, API contracts, date filtering, and performance measurements. References and requirements are clarified in related documentation. [[1]](diffhunk://#diff-4bb29eb8dcedfe22269f84c86e68365ef3bf9a21d3e8eaa0255dc21a175e9f99R1-R18) [[2]](diffhunk://#diff-2e7945765f5f317de775a5e9f47a8254cfebec3e7e2cee97795174798904f36eR1-R15) [[3]](diffhunk://#diff-591c3547e25fbb052e15d1c94b8873677fe73feacef5e8dd1c1a581a53cdb60bR16) [[4]](diffhunk://#diff-ee32781cd7609795577ebd095ebc5cf3708241ab858ffa48c3933f20e2e424b5R13)

### Testing

- **Expanded and improved test coverage:** Tests now verify attribute presence, exclusion of UI-only fields, correct handling of date bounds (including timezone normalization and reversed ranges), and access control enforcement. [[1]](diffhunk://#diff-5cd65ac1cc75e240aa6271e6aa79a04648ddd33b136441c891059948965738e8L153-R153) [[2]](diffhunk://#diff-5cd65ac1cc75e240aa6271e6aa79a04648ddd33b136441c891059948965738e8L188-R189) [[3]](diffhunk://#diff-5cd65ac1cc75e240aa6271e6aa79a04648ddd33b136441c891059948965738e8L205-R207) [[4]](diffhunk://#diff-5cd65ac1cc75e240aa6271e6aa79a04648ddd33b136441c891059948965738e8L247-R282)

### OpenAPI Specification

- **Updated OpenAPI schema:** Documents both the new Assess export endpoint and the enhanced admin export endpoint, including parameters, responses, and error conditions. [[1]](diffhunk://#diff-365143c9eaa6771777c296b844a7d6fcba740f8914dd4c0a43b6ee11b9ab9daaR6138-R6169) [[2]](diffhunk://#diff-365143c9eaa6771777c296b844a7d6fcba740f8914dd4c0a43b6ee11b9ab9daaL8689-R8743)

### CODE REVIEW

File: src/core/tests/application/admin_console/operations/test_admin_api.py [hunk 1]
```diff
@@ -236,8 +237,8 @@ def test_export_stories_and_metadata(client, full_story, api_header, auth_header
     assert news_item_ids.issubset(exported_news_item_ids)
 
     # Attribute we set should be present
-    # attrs = {a.get("key"): a.get("value") for a in data[0].get("attributes", [])}
-    # assert attrs.get("status") == "updated"
```
Comment (not4Pedro) [inline src/core/tests/application/admin_console/operations/test_admin_api.py]: is this to be removed?

File: src/core/core/service/story.py [hunk 2]
```diff
@@ -151,10 +154,44 @@ def export_with_metadata(cls, time_from: datetime | None, time_to: datetime | No
...
+            selectinload(Story.news_items).selectinload(NewsItem.attributes),
+            selectinload(Story.news_items).selectinload(NewsItem.tags),
+            selectinload(Story.news_items).selectinload(NewsItem.osint_source),
+        )
+        return list(db.session.scalars(query).unique().all())
+
+    @classmethod
+    def export_selected(cls, story_ids: list[str], user: User) -> Response | tuple[dict[str, str], int]:
+        story_ids = list(dict.fromkeys(story_ids))
+        if not story_ids or any(not story_id for story_id in story_ids):
```
Comment (not4Pedro) [inline src/core/core/service/story.py]: Can we simplify?
```suggestion
        if not all(story_ids):
```

**Review by not4Pedro — COMMENTED** [review]
It seems the new endpoint is intended to also export multiple stories, however, the UI doesn't allow this as the share button is not among the controls at the top.

Importing the report tag with a story can create a dangling attribute.

I see, that the import path is expected to be used via the admin settings, however, unsure if this is in scope, as when you use the story export and then import using the "create new item" in assess, it succeeds but without importing the data. It is at least a bit confusing.

**Comment by b3n4kh** [comment]
> It seems the new endpoint is intended to also export multiple stories, however, the UI doesn't allow this as the share button is not among the controls at the top.
> 
> Importing the report tag with a story can create a dangling attribute.

The report attribute should probably be dropped while exporting. I will have a look at this

> I see, that the import path is expected to be used via the admin settings, however, unsure if this is in scope, as when you use the story export and then import using the "create new item" in assess, it succeeds but without importing the data. It is at least a bit confusing.

No admin import is supposed to be used after an admin export. Is there anything in the documentation suggesting this? If yes it is there by mistake.

**Comment by not4Pedro** [comment]
> > It seems the new endpoint is intended to also export multiple stories, however, the UI doesn't allow this as the share button is not among the controls at the top.
> > Importing the report tag with a story can create a dangling attribute.
> 
> The report attribute should probably be dropped while exporting. I will have a look at this

okay!

> > I see, that the import path is expected to be used via the admin settings, however, unsure if this is in scope, as when you use the story export and then import using the "create new item" in assess, it succeeds but without importing the data. It is at least a bit confusing.
> 
> No admin import is supposed to be used after an admin export. Is there anything in the documentation suggesting this? If yes it is there by mistake.

Okay, I looked into our documentation, but I think it's lacking in this area. So is this the expected behaviour?

- Admin export -> Admin import is used to load items to another Taranis instance
- Story Assess export -> Admin import is used to load items to another Taranis instance
- Manual news item -> require a JSON of a news item

**Comment by b3n4kh** [comment]
> Okay, I looked into our documentation, but I think it's lacking in this area. So is this the expected behaviour?
> 
> * Admin export -> Admin import is used to load items to another Taranis instance
> * Story Assess export -> Admin import is used to load items to another Taranis instance
> * Manual news item -> require a JSON of a news item

I think this PR is a good point to definetly revist this define at least a minaml sane groundtruth test and document it.

My suggestion would be a two stage approach:

Stage I:

Assess export allows importing to "Create Newsitem" aka "Assess import" AND Admin import.

Admin export allows importing in Admin import and "Assess import".

Eveything else is definetly weird.


Stage II:

We revist this to look into the "NewsItem" vs "Story" exports, and define what exact behaviour we would want in regards to export and importing newsitems, and document this inform the user what will be possible if  not allowing all 4 options as described above for Stories for Newsitems.

**Comment by b3n4kh** [comment]
I now added tests and applied a small fix to ensure Stage I works.

What we need in addition to this should be defined in: #1118

**Review by not4Pedro — APPROVED** [review]
